tpm2/test/tpm2_test.go

```diff
@@ -579,138 +579,55 @@ func TestCertifyEx(t *testing.T) {
 	rw := openTPM(t)
 	defer rw.Close()
 
-	params := Public{
-		Type:       AlgRSA,
-		NameAlg:    AlgSHA256,
-		Attributes: FlagSignerDefault,
-		RSAParameters: &RSAParams{
-			Sign: &SigScheme{
-				Alg:  AlgRSASSA,
-				Hash: AlgSHA256,
-			},
-			KeyBits: 2048,
-		},
-	}
-	signerHandle, signerPub, err := CreatePrimary(rw, HandleOwner, pcrSelection7, emptyPassword, defaultPassword, params)
-	if err != nil {
-		t.Fatalf("CreatePrimary(signer) failed: %s", err)
-	}
-	defer FlushContext(rw, signerHandle)
-
-	subjectHandle, subjectPub, err := CreatePrimary(rw, HandlePlatform, pcrSelection7, emptyPassword, defaultPassword, params)
-	if err != nil {
-		t.Fatalf("CreatePrimary(subject) failed: %s", err)
-	}
-	defer FlushContext(rw, subjectHandle)
-
-	attest, sig, err := CertifyEx(rw, defaultPassword, defaultPassword, subjectHandle, signerHandle, nil, SigScheme{Alg: AlgRSASSA, Hash: AlgSHA256, Count: 0})
-	if err != nil {
-		t.Errorf("Certify failed: %s", err)
-		return
-	}
-
-	attestHash := sha256.Sum256(attest)
-	if err := rsa.VerifyPKCS1v15(signerPub.(*rsa.PublicKey), crypto.SHA256, attestHash[:], sig); err != nil {
-		t.Errorf("Signature verification failed: %v", err)
+	restrictedKeySignerFlags := FlagSignerDefault
+	unrestrictedKeySignerFlags := FlagSign | FlagFixedTPM | FlagFixedParent | FlagSensitiveDataOrigin | FlagUserWithAuth
+	testFlags := []struct {
+		description  string
+		attributes   KeyProp
+		keyScheme    *SigScheme
```

Comment:
nit: It's probably easier to read if `keyScheme` is a value (just type SigScheme) here and you take the address of it with &keyScheme on line 608, where you need to pass a *SigScheme to params.RSAParameters.Sign